### PR TITLE
[Flatpak SDK] Integrate extra Mesa extension

### PR DESCRIPTION
**Pull request by philn**
#### e2649459b018c1966c0a6730e7fa33530e36d8e5
<pre>
[Flatpak SDK] Integrate extra Mesa extension
<a href="https://bugs.webkit.org/show_bug.cgi?id=252534">https://bugs.webkit.org/show_bug.cgi?id=252534</a>

Reviewed by Adrian Perez de Castro.

Provide a flatpak repo including the extra Mesa extension. Inspired from:
- <a href="https://gitlab.gnome.org/GNOME/gnome-build-meta/-/merge_requests/1886">https://gitlab.gnome.org/GNOME/gnome-build-meta/-/merge_requests/1886</a>
- <a href="https://gitlab.gnome.org/GNOME/gnome-build-meta/-/merge_requests/1890">https://gitlab.gnome.org/GNOME/gnome-build-meta/-/merge_requests/1890</a>

The FDO junction was also bumped, because the previous one was shipping a broken Mesa. We didn&apos;t
notice before because we were installing the Mesa extension from Flathub. Now we will have increased
control and reproducibility on this extension.

* Tools/buildstream/Makefile:
* Tools/buildstream/elements/flatpak-platform-extensions-extra.bst: Copied from Tools/buildstream/elements/flatpak-platform-extensions.bst.
* Tools/buildstream/elements/flatpak-platform-extensions.bst:
* Tools/buildstream/elements/flatpak/platform.bst:
* Tools/buildstream/elements/flatpak/sdk.bst:
* Tools/buildstream/project.conf:

Canonical link: <a href="https://commits.webkit.org/260528@main">https://commits.webkit.org/260528@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/71ceae1aad8e52c3472321c6342fa092f5e464d6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108523 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17620 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41373 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117629 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/117827 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/112409 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19071 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8898 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100751 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114290 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14294 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97522 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42262 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/96272 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29162 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/83977 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/97663 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10430 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30511 "Passed tests") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/38/builds/98546 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/8549 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11187 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7420 "Passed tests") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/38/builds/98546 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16577 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50107 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/106089 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12775 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26286 "Passed tests") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3964 "Built successfully and passed tests") | | | |

### CODE REVIEW

File: Tools/buildstream/elements/sdk/icecc.bst [hunk 1]
```diff
@@ -9,6 +9,7 @@ depends:
 - freedesktop-sdk.bst:components/libcap-ng.bst
 - freedesktop-sdk.bst:components/libarchive.bst
 - freedesktop-sdk.bst:components/lzo.bst
+- freedesktop-sdk.bst:components/openssl.bst
```
Comment (aperezdc) [inline Tools/buildstream/elements/sdk/icecc.bst:12]: Adding a dependency to the `icecc` build recipe seems unrelated to the rest of the patch. Let's remove this change before landing, then add it in a separate patch.

**Review by aperezdc — APPROVED** [review]
LGTM, but please do check the comment before landing.

File: Tools/buildstream/elements/sdk/icecc.bst (hunk 1 above)
Comment (philn) [reply Tools/buildstream/elements/sdk/icecc.bst:12]: It is related to the junction bump... upstream SDK updated their libarchive version and now without this icecc configure fails to find libssl and libcrypto.

Comment (aperezdc) [reply Tools/buildstream/elements/sdk/icecc.bst:12]: Got it, thanks for commenting on this.
